provisioning/utils.go

```diff
@@ -64,36 +64,31 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
 	return pod.Status.HostIP, nil
 }
 
-func getServerInternalIP(osclient osclientset.Interface) (string, error) {
+func getServerInternalIPs(osclient osclientset.Interface) ([]string, error) {
 	infra, err := osclient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
 	if err != nil {
 		err = fmt.Errorf("Cannot get the 'cluster' object from infrastructure API: %w", err)
-		return "", err
+		return nil, err
 	}
 	// FIXME(dtantsur): handle the new APIServerInternalIPs field and the dualstack case.
 	switch infra.Status.PlatformStatus.Type {
 	case osconfigv1.BareMetalPlatformType:
-		return infra.Status.PlatformStatus.BareMetal.APIServerInternalIP, nil
+		return infra.Status.PlatformStatus.BareMetal.APIServerInternalIPs, nil
 	case osconfigv1.OpenStackPlatformType:
-		return infra.Status.PlatformStatus.OpenStack.APIServerInternalIP, nil
+		return infra.Status.PlatformStatus.OpenStack.APIServerInternalIPs, nil
 	case osconfigv1.VSpherePlatformType:
-		return infra.Status.PlatformStatus.VSphere.APIServerInternalIP, nil
-	case osconfigv1.AWSPlatformType:
-		return "", nil
-	case osconfigv1.AzurePlatformType:
-		return "", nil
-	case osconfigv1.GCPPlatformType:
```

Comment:
These 3 should be handled the same way as None, not removed

Reply:
Done